File: backend/src/main/java/reports/reports/domain/Subject_.java
package reports.reports.domain;

import javax.persistence.metamodel.ListAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@StaticMetamodel(Subject.class)
public abstract class Subject_ {

    public static volatile SingularAttribute<Subject, Integer> id;
    public static volatile SingularAttribute<Subject, String> name;
    public static volatile SingularAttribute<Subject, String> description;
    public static volatile SingularAttribute<Subject, Long> createdDate;
    public static volatile SingularAttribute<Subject, Long> lastModifiedDate;
    public static volatile SingularAttribute<Subject, String> createdBy;
    public static volatile SingularAttribute<Subject, String> lastModifiedBy;

    public static volatile ListAttribute<Subject, Laboratory> laboratories;

    public static volatile SingularAttribute<Subject, FieldOfStudy> fieldOfStudy;
}

File: frontend/src/main/frontend/src/app/component/subject/subject.ts
import {FieldOfStudy} from "../fieldofstudy/fieldofstudy";
import {Laboratory} from "../laboratory/laboratory";

export class Subject {

  id: number;
  name: string;
  description: string;
  createdDate: string;
  lastModifiedDate: string;
  createdBy: string;
  lastModifiedBy: string;
  laboratories: Laboratory[];
  fieldOfStudy: FieldOfStudy;

  constructor(json? : any) {
    if (json != null) {
      this.id = json.id;
      this.name = json.description;
      this.description = json.description;
      this.createdDate = json.createdDate;
      this.lastModifiedDate = json.lastModifiedDate;
      this.createdBy = json.createdBy;
      this.lastModifiedBy = json.lastModifiedBy;
      if (json.laboratories != null) {
        this.laboratories = Laboratory.toArray(json.laboratories);
      }
      this.fieldOfStudy = json.fieldOfStudy;
    }
  }

  // Utils
  static toArray(jsons : any[]) : Subject[] {
    let subjects : Subject[] = [];
    if (jsons != null) {
      for (let json of jsons) {
        subjects.push(new Subject(json));
      }
    }
    return subjects;
  }
}
